engine/baml-runtime/src/eval_expr.rs

```diff
@@ -0,0 +1,488 @@
+use anyhow::Context;
+use futures::channel::mpsc;
+use futures::stream::{self as stream, StreamExt};
+use internal_baml_core::internal_baml_diagnostics::SerializedSpan;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::{BamlRuntime, FunctionResult};
+use baml_types::expr::{Expr, ExprMetadata, Name};
+use baml_types::Arrow;
+use baml_types::{BamlMap, BamlValue, BamlValueWithMeta};
+use internal_baml_core::ir::repr::IntermediateRepr;
+
+pub struct EvalEnv<'a> {
```

Comment:
i assume this is $$\Gamma$$ aka the typing context - but how are typing contexts stacked?